includes/course-theme/class-sensei-course-theme-styles.php

```diff
@@ -0,0 +1,247 @@
+<?php
+/**
+ * File containing Sensei_Course_Theme_Styles class.
+ *
+ * @package sensei-lms
+ * @since   $$next-version$$
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Add support for global styles colors.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_Course_Theme_Styles {
+
+	/**
+	 * Add hooks.
+	 */
+	public static function init() {
+		add_action( 'wp_head', [ self::class, 'output_global_styles_colors' ] );
+		add_action( 'render_block', [ self::class, 'apply_block_support' ], 10, 2 );
+	}
+
+	/**
+	 * Add block colors as inline style with CSS variables.
+	 *
+	 * @access private
+	 *
+	 * @param string $block_content Rendered block content.
+	 * @param array  $block         Block object.
+	 *
+	 * @return string Filtered block content.
+	 */
+	public static function apply_block_support( $block_content, $block ) {
+		if ( ! $block_content ) {
+			return $block_content;
+		}
+
+		$colors = self::get_block_colors( $block['attrs'] );
+
+		if ( empty( $colors ) ) {
+			return $block_content;
+		}
+
+		$style = esc_attr( implode( ' ', $colors ) );
+		return self::set_style_attribute( $block_content, $style );
+
+	}
+
+	/**
+	 * Get global styles colors and output them as CSS variables.
+	 *
+	 * @access private
+	 */
+	public static function output_global_styles_colors() {
+		$styles = wp_get_global_styles();
+		$vars   = self::get_colors_as_css_variables( $styles );
+
+		if ( ! empty( $vars ) ) {
+			self::output_style( implode( "\n", $vars ) );
+		}
+	}
+
+	/**
+	 * Get colors from a styles configuration and generate CSS variables.
+	 *
+	 * @param array $styles Styles object.
+	 *
+	 * @return array Name-value pairs of CSS variables
+	 */
+	private static function get_colors_as_css_variables( $styles ) {
+
+		if ( empty( $styles ) ) {
+			return [];
+		}
+
+		$colors = self::get_colors( $styles );
+		return self::format_css_variables( $colors, 'sensei-' );
+	}
+
+	/**
+	 * Get relevant colors from global styles or a block's attributes.
+	 *
+	 * @param array $styles Styles object.
+	 *
+	 * @return array
+	 */
+	private static function get_colors( $styles ) {
+
+		if ( empty( $styles ) ) {
+			return [];
+		}
+
+		$vars = [];
+
+		$element_colors = $styles['color'] ?? $styles['elements']['color'] ?? null;
+
+		$vars['text-color']             = $element_colors['text'] ?? $styles['textColor'] ?? null;
+		$vars['background-color']       = $element_colors['background'] ?? $styles['backgroundColor'] ?? null;
+		$vars['primary-contrast-color'] = $vars['--sensei-background-color'];
```

Comment:
From what I understand the `$vars['--sensei-background-color']` is empty at this point.